### PR TITLE
fixed user registration bug when email login is enabled

### LINKED ISSUE #5031: Create new user fails if login with email is enabled
The schema of the 'user creation form' should not include the username as this is not allowed.
https://6.docs.plone.org/plone.restapi/docs/source/endpoints/users.html#create-user

Volto 16 and 17

### PR DESCRIPTION
**Pull request by mehedikhan72**
Fixes #5031 
When email was set to be the username field, the actual username field still appeared and the registration was broken.

I fixed it by showing the username field conditionally based on whether this setting was enabled.

I used the following to get our desired boolean value and stored it in a state, which was used to conditionally render the username input element.
```
  checkLoginUsingEmailStatus = async () => {
    await this.props.getControlpanel('security');
    this.setState({
      loginUsingEmail: this.props.controlPanelData.data.use_email_as_login,
    });
  };
```
  
  Feel free to let me know if I need to make changes regarding this PR. Thanks!

### CODE REVIEW

File: .gitignore
Comment (stevepiercy) [inline .gitignore:22]: No, we need `yarn.lock` in the repository to manage dependencies. Please remove this line.

```suggestion
```

Instead, do not make changes to `yarn.lock`, and do not commit changes to it. You can add it to your user's `.gitignore` instead of the project's, if it really bothers you.

File: yarn.lock
Comment (stevepiercy) [inline yarn.lock:11926]: Revert all changes to this file and do not commit them in your pull request.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Comment by mehedikhan72** [comment]
Hi @stevepiercy , I made the changes you requested but these checks are still appearing as unsuccessful. What should I do next?

**Comment by mehedikhan72** [comment]
Looks like there are still issues with yarn. I'm not sure why. I copy pasted the entire file from volto's master branch to my project to make sure that they match and then made a commit and after that I wrote the following to ignore yarn.lock in my future commits.

```
git update-index --skip-worktree yarn.lock
```

**Comment by ksuess** [comment]
> Looks like there are still issues with yarn. I'm not sure why. I copy pasted the entire file from volto's master branch to my project to make sure that they match …

Would you please just copy the master branch yarn.lock and paste it in your branch once more. There is a weird character and a missing new line in the end. With an then unchanged yarn.lock the test runs should proceed one step further.

Afterwards you can run on your local machine `yarn test`. You will see an error on `checkLoginUsingEmailStatus` 
Maybe check for `this.props.controlPanelData`before setting the state?

**Comment by mehedikhan72** [comment]
Nice! @ksuess thanks for the suggestions. Everything seems to be okay now. When can I expect this to be merged?

On another note, what version of prettier does Plone need? I ran it but still the check wasn't successful. Then I ran eslint and it was okay afterwards.

File: news/5031.bugfix
Comment (stevepiercy) [inline news/5031.bugfix:1]: For future contributions:

- A commit message should say what it actually fixes, not just refer to an issue number.
- We already know it is a bug fix by its extension of `.bugfix` so saying it is a bug fix in the message is redundant. Towncrier automatically renders sections as shown in this example: https://6.docs.plone.org/volto/release-notes/index.html#alpha-22-2023-07-28
- Your GitHub handle should be the last part of a change log entry.

```suggestion
Allow a user to register when they use an email address as their username. [#5031](https://github.com/plone/volto/issues/5031) @mehedikhan72
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
A maintainer should review this.

**Comment by stevepiercy** [comment]
@mehedikhan72 would you please edit your original issue description from `#5031` to `Fixes #5031`? Doing so will then automatically close that issue when this pull request is merged. Alternatively, you can reference issues that should be closed upon merge in the right hand side of GitHub under "Development". Thank you!

**Review by stevepiercy — APPROVED** [review]
Code needs review from maintainer.

**Comment by mehedikhan72** [comment]
> 

@stevepiercy  Changes were made. Thank you for your guidance throughout this PR.

**Review by davisagli — APPROVED** [review]
Thanks for the fix! I tested and it seems to work well.

**Comment by mehedikhan72** [comment]
My pleasure @davisagli !

**Pull request merged**
